connect-inject/health_check_resource.go

```diff
@@ -0,0 +1,311 @@
+package connectinject
+
+import (
+	ctx "context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/go-hclog"
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	// labelInject is the label which is applied by the connect-inject webhook to all pods.
+	// This is the key the controller will use on the label filter for its lister, watcher and reconciler.
+	labelInject = "consul.hashicorp.com/connect-inject-status"
+
+	// kubernetesSuccessReasonMsg will be passed for passing health check's Reason to Consul
+	kubernetesSuccessReasonMsg = "Kubernetes Health Checks Passing"
+
+	// passing/failing strings passed into Consul health check as Status
+	healthCheckPassing  = "passing"
+	healthCheckCritical = "critical"
+)
+
+type HealthCheckResource struct {
+	Log          hclog.Logger
+	Clientset    kubernetes.Interface
+	ClientConfig *api.Config
+
+	// ConsulPort will be 8500/8501 based on TLS enabled
+	ConsulPort string
```

Comment:
could we instead make this `TLSEnabled` bool since that is what it's used for?

Reply:
Yup, I've refactored it to skip both the port and TLS (coming soon to a commit near you)